test/testdata/infer/private_methods.rb

```diff
@@ -62,8 +62,8 @@ def calling_private_in_parent
 
 # TODO: The following methods should also error out. They currently do not do that due to issues with the instrinsics.
 Test.new.splat_call(*T.unsafe(nil)) # Currently no Error, since T.unsafe makes Magic_callWithSplat#apply return prematurely due to the untyped argument.
-Test.new.splat_and_block_call(*[1, 'a'], &nil) # Currently no Error, since the nil block makes Magic_callWithSplatAndBlock return prematurely.
-Test.new.block_call(&nil) # Currently no Error, since the nil block makes Magic_callWithBlock return prematurely.
+Test.new.splat_and_block_call(*[1, 'a'], &nil) # error: Non-private call to private method `splat_and_block_call` on `Test`
+Test.new.block_call(&nil) # error: Non-private call to private method `block_call` on `Test`
```

Comment:
Very non-blocking suggestion: you might tweak the comment here, since the second and third calls now _are_ erroring out.